packages/core/src/pages/[slug]/p.tsx

```diff
@@ -51,6 +51,22 @@ function Page({ product, sections, globalSections }: Props) {
   const description = seo.description || storeConfig.seo.description
   const canonical = `${storeConfig.storeUrl}${seo.canonical}`
 
+  const getStructuredDataOffers = () => {
```

Comment:
do you think does it worth to move this logic to the `getStaticProps` and return the result as prop?

Reply:
+1